src/main/java/org/testng/IDataProviderMethod.java

```diff
@@ -0,0 +1,41 @@
+package org.testng;
+
+import java.lang.reflect.Method;
+import java.util.List;
+
+/**
+ * Represents the attributes of a {@link org.testng.annotations.DataProvider} annotated method.
+ */
+public interface IDataProviderMethod {
```

Comment:
I decided to stick to the conventions that all the current TestNG listeners are following. None of them are exposing any sub-classes of `org.testng.annotations.IAnnotation`. So I followed the same convention.